core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -651,60 +651,126 @@ class UserService(protected val ctx: RawlsRequestContext,
       }
     }
 
+  private def getLegacyBillingPolicies(samRole: ProjectRole): Seq[SamResourcePolicyName] =
+    samRole match {
+      case ProjectRoles.Owner => Seq(SamBillingProjectPolicyNames.owner)
+      case ProjectRoles.User =>
+        Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
+    }
+  private def getV2BillingPolicy(samRole: ProjectRole): SamResourcePolicyName =
+    samRole match {
+      case ProjectRoles.Owner => SamBillingProjectPolicyNames.owner
+      case ProjectRoles.User  => SamBillingProjectPolicyNames.workspaceCreator
+    }
+
   def addUserToBillingProject(projectName: RawlsBillingProjectName,
                               projectAccessUpdate: ProjectAccessUpdate
   ): Future[Unit] =
     requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
-      val policies = projectAccessUpdate.role match {
-        case ProjectRoles.Owner => Seq(SamBillingProjectPolicyNames.owner)
-        case ProjectRoles.User =>
-          Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
+      val policies = getLegacyBillingPolicies(projectAccessUpdate.role)
+      addUserToBillingProjectInner(projectName, projectAccessUpdate, policies)
+    }
+
+  private def addUserToBillingProjectInner(projectName: RawlsBillingProjectName,
```

Comment:
If you find it useful, you can nest this inside the main function.
Personally, I tend to prefer nesting when I'm breaking the function up for reasons that don't involve re-use, since it makes it clear where the logic belongs, and you still have the context for values that are just pass-through.

Totally optional though.

Reply:
This is used by 2 functions-- the v1 and v2 wrappers.

Reply:
Ignore me, then :-P